### DISCUSSION TITLE
Use github token with my github page

### DISCUSSION THREAD

**Discussion by YerkoPalma** [community/community#21535] [Other Feature Feedback, Questions, & Ideas]
<p>I’m creating a github page and I want that with some user interactions, some info get saved to a github repo. </p>
<p>To do this, I’m using <a href="https://github.com/github-tools/github" rel="nofollow noopener">github api</a> module, but when I submitted the commit with the changes, Github deprecated my token for security reasons. </p>
<p>I’m ok with that, it totally makes sense. But now, I’m confused. How am I supposed to use an access token in a github pages repo? should I set it as a environment variable and not commit it? but if it is in a github pages repo I have to commit it. Or is there other way??</p>
<p>I’m using browserify and other npm modules with no site generator. Any thanks is appreciated.</p>

**Comment by foopod** [comment, marked as answer]
<p>I can’t think of a way to do this with Github Pages alone.</p>
<p>I am in no way affiliated, but use Netlify to do exactly this (Bonus, it’s free).</p>
<p>Either using Snippet Injection (<a href="https://www.netlify.com/docs/inject-analytics-snippets/" rel="nofollow noopener">https://www.netlify.com/docs/inject-analytics-snippets/</a>) if it is pure html/css/js.</p>
<p>Or using Environment Variables that get included in the build process for static site generation (<a href="https://www.netlify.com/docs/build-settings/" rel="nofollow noopener">https://www.netlify.com/docs/build-settings/</a>).</p>

**Comment by YerkoPalma** [comment]
<p>Thanks for the suggestion, will check it asap.</p>
